src/Fixer/Operator/OperatorLinebreakFixer.php

```diff
@@ -0,0 +1,229 @@
+        '==',
+        '>=',
+        '===',
+        '!=',
+        '<>',
+        '!==',
+        '<=',
+        '-=',
+        '%=',
+        '*=',
+        '|=',
+        '+=',
+        '<<',
+        '<<=',
+        '>>',
+        '>>=',
+        '^=',
+        '**',
+        '**=',
+        '<=>',
+        '??',
+    ];
+
+    /** @var string[] */
+    private $operators;
+
+    /** @var string */
+    private $position = 'beginning';
+
+    public function __construct()
+    {
+        $this->operators = \array_merge(self::BOOLEAN_OPERATORS, self::NON_BOOLEAN_OPERATORS);
+
+        parent::__construct();
+    }
+
+    /**
+     * @return FixerDefinition
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Binary operators must always be at the beginning or at the end of the line.',
+            [new CodeSample('<?php
+function foo() {
+    return $bar ||
+        $baz;
+}
+')]
+        );
+    }
+
+    /**
+     * @return FixerConfigurationResolver
+     */
+    public function getConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('only_booleans', 'whether to limit operators to only boolean ones'))
+                ->setDefault(false)
+                ->setAllowedTypes(['bool'])
+                ->getOption(),
+            (new FixerOptionBuilder('position', 'whether to place operators at the beginning or at the end of the line'))
+                ->setDefault($this->position)
+                ->setAllowedValues(['beginning', 'end'])
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        if (isset($configuration['only_booleans']) && $configuration['only_booleans']) {
+            $this->operators = self::BOOLEAN_OPERATORS;
+        }
+        $this->position = isset($configuration['position']) ? $configuration['position'] : $this->position;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        return 0;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        if ('beginning' === $this->position) {
+            $this->fixMoveToTheBeginning($tokens);
+        } else {
+            $this->fixMoveToTheEnd($tokens);
+        }
+    }
+
+    private function fixMoveToTheBeginning(Tokens $tokens)
+    {
+        for ($index = 0; $index < $tokens->count(); ++$index) {
+            $tokenContent = \strtolower($tokens[$index]->getContent());
+
+            if (!\in_array($tokenContent, $this->operators, true)) {
```

Comment:
So uhm, isn't the runtime of this one fixer going to be enormous, doing an `in_array` for each and every token? I propose either making the operators the keys, or doing an `array_flip` to achive the same, and then here do either `array_key_exists` or `isset`.

This obviously holds for `fixMoveToTheEnd` too.